libraries/botbuilder-dialogs/botbuilder/dialogs/prompts/oauth_prompt.py

```diff
@@ -384,9 +401,78 @@ def __init__(
                         Activity(type="invokeResponse", value=InvokeResponse(404))
                     )
             except Exception:
-                context.send_activity(
+                await context.send_activity(
                     Activity(type="invokeResponse", value=InvokeResponse(500))
                 )
+        elif self._is_token_exchange_request_invoke(context):
+            if isinstance(context.activity.value, dict):
+                context.activity.value = TokenExchangeInvokeRequest().from_dict(
+                    context.activity.value
+                )
+            if not (
+                context.activity.value
+                and self._is_token_exchange_request(context.activity.value)
+            ):
+                await context.send_activity(
+                    self._get_token_exchange_invoke_response(
+                        400,
+                        "The bot received an InvokeActivity that is missing a TokenExchangeInvokeRequest value."
+                        " This is required to be sent with the InvokeActivity.",
+                    )
+                )
+            elif (
+                context.activity.value.connection_name != self._settings.connection_name
+            ):
+                await context.send_activity(
+                    self._get_token_exchange_invoke_response(
+                        400,
+                        "The bot received an InvokeActivity with a TokenExchangeInvokeRequest containing a"
+                        " ConnectionName that does not match the ConnectionName expected by the bots active"
+                        " OAuthPrompt. Ensure these names match when sending the InvokeActivityInvalid"
+                        " ConnectionName in the TokenExchangeInvokeRequest",
+                    )
+                )
+            elif not getattr(context.adapter, "exchange_token"):
+                await context.send_activity(
+                    self._get_token_exchange_invoke_response(
+                        502,
```

Comment:
In other languages we have status code constants (in javascript StatusCodes) so for example you can do `StatusCodes.OK`, etc. Please add that in python if not already existing, or use it if it already exists.

Reply:
Corrected